### PR TITLE
Leaf/c1.1 swiping mechanism ux

### PR DESCRIPTION
**Pull request by Darren52BT**
Addresses #87.
Implements the design of the profile cards and the swiping mechanism for the profile cards.  You should be able to swipe right and left to traverse through the cards both on mobile and mouse. You are also able to swipe through clicking the arrow buttons on the bottom of the screen, as well as press the right arrow key / 'd' key to go right and the left arrow key / 'a' key to go left. When the screen width gets smaller than 700 pixels in width the cards should appear more closely stacked together and the main card should take up more width. When the screen width goes below 500 pixels OR the screen height goes below 630 pixels much of the card's content loses its padding, causing the card to become shorter in height. 
image

### CODE REVIEW

File: meshapp/src/ProfileCardCarousel/Swiper.tsx [hunk 1]
```diff
@@ -0,0 +1,219 @@
+import React, { useCallback, useEffect } from "react";
+import { useState } from "react";
```
Comment (rezatnoMsirhC) [inline meshapp/src/ProfileCardCarousel/Swiper.tsx:2]: Why aren't these combined into `import { React, useCallback, useEffect, useState } from "react";` instead?

File: meshapp/src/ProfileCardCarousel/Swiper.tsx [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+        "Looking for a Mentor",
+      ],
+    },
+  ];
+
+  //keeps track of the current card we're on
+  const [position, positionSet] = useState(0);
+
+  //moves to the next card if possible
+  const onRight = () => {
```
Comment (rezatnoMsirhC) [inline meshapp/src/ProfileCardCarousel/Swiper.tsx:82]: Please rename this and `onLeft` to `goRight` and `goLeft`. Refer to [this article on semantic method-naming](http://codelegance.com/semantic-method-naming/) for some more insight on the motivation for naming functions this way.

File: meshapp/src/ProfileCardCarousel/Swiper.tsx [hunk 3]
```diff
@@ -0,0 +1,219 @@
+import React, { useCallback, useEffect } from "react";
+import { useState } from "react";
+import { useSwipeable } from "react-swipeable";
+import { Grid, IconButton } from "@mui/material";
+import profileCard from "./ProfileCard";
+import ArrowBackIcon from "@mui/icons-material/ArrowBack";
+import ArrowForwardIcon from "@mui/icons-material/ArrowForward";
+
+function Swiper() {
```
Comment (rezatnoMsirhC) [inline meshapp/src/ProfileCardCarousel/Swiper.tsx:9]: This is a pretty large function. Can `Swiper` be converted to a class with smaller methods instead?

File: meshapp/src/ProfileCardCarousel/Swiper.tsx [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+  let divs = profiles.map(
+    ({ image, name, job, biography, interests, details }, index: number) => {
+      //controls which card shows up on top of the stack
+      let onTop: number = 0;
+      //controls the positioning of the cards depending on what index they are from the current position
+      //if window width is lower than 700 it will stack the cards closer together
+      let leftPos: string =
+        windowWidth > 700
+          ? `${(index - position) * 15 + 14}vw`
+          : `${(index - position) * 10 + 4.5}vw`;
```
Comment (rezatnoMsirhC) [inline meshapp/src/ProfileCardCarousel/Swiper.tsx:141]: There are too many magic-numbers here. I suggest converting these into constants to make your code more readable and easier to update later (i.e. `private static readonly MAX_WINDOW_WIDTH : number = 700;`).

File: meshapp/src/index.tsx [hunk 5]
```diff
@@ -1,15 +1,15 @@
...
 import TwoFactorAuthReminders from "./two-factor-auth/two-factor-reminder";
 
 import LoggedOutPage from "./home/logged-out/home";
-
+import Slider from "./ProfileCardCarousel/Swiper";
 const root = ReactDOM.createRoot(document.getElementById("root")!);
 root.render(
   <React.StrictMode>
-    <TwoFactorAuthReminders />
+    {/* <TwoFactorAuthReminders /> */}
```
Comment (rezatnoMsirhC) [inline meshapp/src/index.tsx:11]: Why is this being removed?

File: .gitignore [hunk 6]
```diff
@@ -3,4 +3,5 @@
 
 # Django
 *.pyc
-*.sqlite3
\ No newline at end of file
+*.sqlite3
+node_modules
```
Comment (rezatnoMsirhC) [inline .gitignore:7]: I think we're already ignoring this in `meshapp/.gitignore`

File: package-lock.json
Comment (rezatnoMsirhC) [inline package-lock.json:1]: Please remove this file

**Review by rezatnoMsirhC — CHANGES_REQUESTED** [review]

File: meshapp/src/index.tsx (hunk 5 above)
Comment (Darren52BT) [reply meshapp/src/index.tsx]: I commented it out for a while just to focus on the profile cards and forgot to put it back it, I added it back

File: .gitignore (hunk 6 above)
Comment (Darren52BT) [reply .gitignore]: I removed it

File: package-lock.json
Comment (Darren52BT) [reply package-lock.json:1]: Done

File: meshapp/src/ProfileCardCarousel/Swiper.tsx (hunk 4 above)
Comment (Darren52BT) [reply meshapp/src/ProfileCardCarousel/Swiper.tsx]: I have converted them to constants

File: meshapp/src/ProfileCardCarousel/Swiper.tsx (hunk 1 above)
Comment (Darren52BT) [reply meshapp/src/ProfileCardCarousel/Swiper.tsx]: My bad I forgot to fix that, should be good now

File: meshapp/src/ProfileCardCarousel/Swiper.tsx (hunk 2 above)
Comment (Darren52BT) [reply meshapp/src/ProfileCardCarousel/Swiper.tsx]: done, ty for the article

File: meshapp/src/ProfileCardCarousel/Swiper.tsx (hunk 3 above)
Comment (Darren52BT) [reply meshapp/src/ProfileCardCarousel/Swiper.tsx]: I've separated several functionalities of Swiper into different functions

**Review by misslame — APPROVED** [review]
Nice Work

**Review by PattyCakesxD — CHANGES_REQUESTED** [review]
Concern: Should the interests be expandable by clicking it? (Theres more than 3 tags)
image

**Review by rezatnoMsirhC — APPROVED** [review]

**Review by PattyCakesxD — APPROVED** [review]
lgtm

**Pull request merged**